plugins/webp-uploads/picture-element.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * Add `picture` element support
+ *
+ * @package webp-uploads
+ *
+ * @since n.e.x.t
+ */
+
+/**
+ * Potentially wrap an image tag in a picture element.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $image         The image tag.
+ * @param string $context       The context for the image tag.
+ * @param int    $attachment_id The attachment id.
+ *
+ * @return string The new image tag.
+ */
+function webp_uploads_wrap_image_in_picture( string $image, string $context, int $attachment_id ): string {
+	if ( 'the_content' !== $context ) {
+		return $image;
+	}
+	$image_meta              = wp_get_attachment_metadata( $attachment_id );
+	$original_file_mime_type = get_post_mime_type( $attachment_id );
+	if ( false === $original_file_mime_type || ! isset( $image_meta['sizes'] ) ) {
+		return $image;
+	}
+
+	// Collect all of the sub size image mime types.
```

Comment:
Suggested by LanguageTool in PhpStorm:

```suggestion
	// Collect all the sub size image mime types.
```